static/js/global.js

```diff
@@ -0,0 +1,212 @@
+function  fillTableCell_Text (cellData) {
+    var td = document.createElement("td");
+    var txt = document.createTextNode(cellData);
+    td.appendChild(txt);
+    return td;
+}
+
+function Forecast() {
+    var self = this;
+
+    this.createForecastTable = function (data) {
+        if (data.hasOwnProperty('error_msg')) {
+            historyPage.loadHistoryPage();
+            $("#forecast_table").hide();
+            $("#error_msg").empty().show();
+
+            var error_div = document.getElementById('error_msg');
+            var txt = document.createTextNode(data['error_msg']);
+            error_div.appendChild(txt);
+
+        } else {
+            $("#error_msg").empty().hide();
+            $("#forecast_table_body").empty();
+            $("#forecast_table").show();
+
+            var table = document.getElementById("forecast_table_body");
+            var rows = document.createElement("tr");
+            rows.appendChild(fillTableCell_Text(data['city'] + data['country']));
+            rows.appendChild(fillTableCell_Text(data['min_temperature']));
+            rows.appendChild(fillTableCell_Text(data['max_temperature']));
+
+            table.appendChild(rows);
+        }
+    };
+
+    this.getForecastById = function (id) {
+        $("#spinner").show();
+        $.ajax({
+            type: 'GET',
+            url: "/forecast/".concat(String(id), "/"),
+            success: function (forecast) {
+                $('#forecast_table_body').empty();
+                self.createForecastTable(forecast);
+                $("#spinner").hide();
+            },
+            error: function (data) {
+                $('#forecast_table_body').empty();
+                $("#spinner").hide();
+                showErrorMsg("Oops..     Request failed: " + data.statusText +
+                ". Please try again later.");
+            }
+        });
+    };
+}
+
+function Pagination (perPageElementId) {
+    this.elementId = perPageElementId;
+    this.currentPage = 0;
+    this.perPage = null;
+    this.maxPage = null;
+    var self  = this;
+
+    this.setPaginationData = function(data){
+        self.currentPage = data['paginator']['current'];
+        self.perPage = data['paginator']['per_page'];
+        self.maxPage = data['paginator']['max_page'];
+    };
+
+    this.getNumberOfElementsOnPage = function () {
+        var number;
+        number = document.getElementById(self.elementId).value;
```

Comment:
The element can be stored in the class field to avoid calling document.getElementById all the time.